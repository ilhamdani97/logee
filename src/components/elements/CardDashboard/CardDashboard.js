import React from 'react';
import PropTypes from 'prop-types';
import style from './styles.scoped.css';
import { Col, Row, Tooltip, Typography } from 'logeeport-ui';
import { Link } from 'react-router-dom';

const CardDashboard = ({ data }) => {
  const { content, title, toolTip,bgColor, link } = data;
  const { direction, message } = toolTip;

  const classes = [style.root,style[bgColor]].join(' ').trim();
  return (
    <Row className={classes}>
      <Col className={style.title} xs="8">
        {link ? <Link to={link}>{title}</Link> : title}</Col>
      <Col className={style.toolTips} xs="4">
        <Tooltip content={ToolTipMsg(message)} direction={direction}>
          <img src="/assets/ic-info.svg"/>
        </Tooltip>
      </Col>
      <Col className={style.content} xs="12">
        {
          link ?
            <Link className={style.linkArrow} to={link}>
              <Typography>{content}</Typography>
              <img src="/assets/ic-arrow-right.png"/>
            </Link> :
            <Typography >{content}</Typography>
        }
      </Col>
    </Row>
  );
};

const ToolTipMsg = (msg) => {
  return(
    <div className={style.tootlMsg}>
      {msg}
    </div>
  );
};

CardDashboard.defaultProps = {
  data: {},
};

CardDashboard.propTypes = {
  data: PropTypes.object,
};

export default CardDashboard;
