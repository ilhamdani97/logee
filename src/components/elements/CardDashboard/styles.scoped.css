.root{
  background-color: #f5f5f5;
  border-radius: 4px;
}
.root.green{
  background-color: var(--green-10);
}

.root.red{
  background-color: var(--red-10);
}



.toolTips{
  text-align: right;
}

.toolTips img + div{
  bottom: -50px !important;
  display: flex;
  align-items: center;
}

.title{
  font-size: 14px;
}

.title a{
  font-size: inherit;
}


.content{
  padding-top: 0;
}
.content p{
  font-size: 24px;
  font-weight: 600;
}

.tootlMsg{
  width: 230px;
  white-space: initial;
  text-align: center;
  font-size: 8px;
}

.tootlMsg p{
  font-size: 12px;
}

.linkArrow{
  display: flex;
  justify-content: space-between;
}

.linkArrow img{
  width: 8px;
  height: 14px;
  margin-top: 8px;
}