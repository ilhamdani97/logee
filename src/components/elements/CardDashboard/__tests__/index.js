import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import CardDashboard from '../CardDashboard';


const data = {
  'title': 'Belum Dibayar',
  'toolTip': {
    'message': {
      'type': 'p',
      'key': null,
      'ref': null,
      'props': {
        'children': [
          'Total transaksi proforma pengurusan baru ',
          {
            'type': 'br',
            'key': null,
            'ref': null,
            'props': {},
            '_owner': null,
            '_store': {}
          },
          'dan perpanjangan yang belum dibayar PPJK'
        ]
      },
      '_owner': null,
      '_store': {}
    },
    'direction': 'left'
  },
  'content': 'Rp 0',
  'bgColor': 'red',
  'link': '/proforma?proformaDetailsStatus=CREATED'
};

const nolink = {
  'title': 'Belum Dibayar',
  'toolTip': {
    'message': {
      'type': 'p',
      'key': null,
      'ref': null,
      'props': {
        'children': [
          'Total transaksi proforma pengurusan baru ',
          {
            'type': 'br',
            'key': null,
            'ref': null,
            'props': {},
            '_owner': null,
            '_store': {}
          },
          'dan perpanjangan yang belum dibayar PPJK'
        ]
      },
      '_owner': null,
      '_store': {}
    },
    'direction': 'left'
  },
  'content': 'Rp 0',
  'bgColor': 'red',
};

describe('src/components/elements/CardDashboard', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<CardDashboard data={data}/>);
    expect(tree).toMatchSnapshot();
  });

  test('nolink', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<CardDashboard data={nolink}/>);
    expect(tree).toMatchSnapshot();
  });

});
