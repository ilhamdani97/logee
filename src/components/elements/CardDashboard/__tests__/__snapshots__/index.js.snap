// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/CardDashboard nolink 1`] = `
<Row
  alignContent=""
  alignItem=""
  className=""
  direction=""
  id=""
  justify=""
  noGutter={false}
>
  <Col
    className=""
    id=""
    lg=""
    md=""
    offsetLg=""
    offsetMd=""
    offsetSm=""
    offsetXL=""
    offsetXs=""
    orderLg=""
    orderMd=""
    orderSm=""
    orderXl=""
    orderXs=""
    sm=""
    xl=""
    xs="8"
  >
    Belum Dibayar
  </Col>
  <Col
    className=""
    id=""
    lg=""
    md=""
    offsetLg=""
    offsetMd=""
    offsetSm=""
    offsetXL=""
    offsetXs=""
    orderLg=""
    orderMd=""
    orderSm=""
    orderXl=""
    orderXs=""
    sm=""
    xl=""
    xs="4"
  >
    <Tooltip
      content={
        <div>
          Object {
            "_owner": null,
            "_store": Object {},
            "key": null,
            "props": Object {
              "children": Array [
                "Total transaksi proforma pengurusan baru ",
                Object {
                  "_owner": null,
                  "_store": Object {},
                  "key": null,
                  "props": Object {},
                  "ref": null,
                  "type": "br",
                },
                "dan perpanjangan yang belum dibayar PPJK",
              ],
            },
            "ref": null,
            "type": "p",
          }
        </div>
      }
      delay={0}
      direction="left"
    >
      <img
        src="/assets/ic-info.svg"
      />
    </Tooltip>
  </Col>
  <Col
    className=""
    id=""
    lg=""
    md=""
    offsetLg=""
    offsetMd=""
    offsetSm=""
    offsetXL=""
    offsetXs=""
    orderLg=""
    orderMd=""
    orderSm=""
    orderXl=""
    orderXs=""
    sm=""
    xl=""
    xs="12"
  >
    <Typography
      spacing=""
    >
      Rp 0
    </Typography>
  </Col>
</Row>
`;

exports[`src/components/elements/CardDashboard render 1`] = `
<Row
  alignContent=""
  alignItem=""
  className=""
  direction=""
  id=""
  justify=""
  noGutter={false}
>
  <Col
    className=""
    id=""
    lg=""
    md=""
    offsetLg=""
    offsetMd=""
    offsetSm=""
    offsetXL=""
    offsetXs=""
    orderLg=""
    orderMd=""
    orderSm=""
    orderXl=""
    orderXs=""
    sm=""
    xl=""
    xs="8"
  >
    <Link
      to="/proforma?proformaDetailsStatus=CREATED"
    >
      Belum Dibayar
    </Link>
  </Col>
  <Col
    className=""
    id=""
    lg=""
    md=""
    offsetLg=""
    offsetMd=""
    offsetSm=""
    offsetXL=""
    offsetXs=""
    orderLg=""
    orderMd=""
    orderSm=""
    orderXl=""
    orderXs=""
    sm=""
    xl=""
    xs="4"
  >
    <Tooltip
      content={
        <div>
          Object {
            "_owner": null,
            "_store": Object {},
            "key": null,
            "props": Object {
              "children": Array [
                "Total transaksi proforma pengurusan baru ",
                Object {
                  "_owner": null,
                  "_store": Object {},
                  "key": null,
                  "props": Object {},
                  "ref": null,
                  "type": "br",
                },
                "dan perpanjangan yang belum dibayar PPJK",
              ],
            },
            "ref": null,
            "type": "p",
          }
        </div>
      }
      delay={0}
      direction="left"
    >
      <img
        src="/assets/ic-info.svg"
      />
    </Tooltip>
  </Col>
  <Col
    className=""
    id=""
    lg=""
    md=""
    offsetLg=""
    offsetMd=""
    offsetSm=""
    offsetXL=""
    offsetXs=""
    orderLg=""
    orderMd=""
    orderSm=""
    orderXl=""
    orderXs=""
    sm=""
    xl=""
    xs="12"
  >
    <Link
      to="/proforma?proformaDetailsStatus=CREATED"
    >
      <Typography
        spacing=""
      >
        Rp 0
      </Typography>
      <img
        src="/assets/ic-arrow-right.png"
      />
    </Link>
  </Col>
</Row>
`;
